tests/test_qaoa_objective_sk.py

```diff
@@ -0,0 +1,213 @@
+        for simulator in simulators_to_run_names_no_qiskit
+    ]
+    print(qaoa_objectives)
+    assert np.allclose(qaoa_objectives, expected_energy)
+
+
+def test_sk_qaoa_convergence_with_p():
+    N = 8
+    J = np.random.randn(N, N)
+    J = (J + J.T) / 2
+    np.fill_diagonal(J, 0)
+
+    obj = partial(sk_obj, J=J)
+    precomputed_energies = precompute_energies(obj, N)
+    optimal_cut = np.max(precomputed_energies)
+
+    last_ar = [0.0, 0.0]
+
+    for p in range(1, 18):
+        gamma, beta = get_sk_gamma_beta(p)
+        for objective in ["expectation"]:
+            qaoa_objectives = [
+                get_qaoa_sk_objective(
+                    N, p, J=J, precomputed_cuts=precomputed_energies, parameterization="gamma beta", simulator=simulator, objective=objective
+                )(gamma / np.sqrt(N), beta)
+                for simulator in simulators_to_run_names_no_qiskit
+            ]
+            print("p: ", p, "Approximation Ratio:", qaoa_objectives / optimal_cut)
+            current_ar = qaoa_objectives / optimal_cut
+            assert list(current_ar) > list(last_ar)
+            last_ar = current_ar
+
+
+def test_sk_withJ_and_maxcut_with_terms():
+    N = 8
+    J = np.random.randn(N, N)
+    J = (J + J.T) / 2
+    np.fill_diagonal(J, 0)
+
+    terms = [(2 * J[spin_pair[0], spin_pair[1]] / np.sqrt(N), spin_pair) for spin_pair in combinations(range(N), r=2)]
+
+    for p in range(1, 18):
+        gamma, beta = get_sk_gamma_beta(p)
+        f1 = [
+            get_qaoa_objective(N, terms=terms, parameterization="gamma beta", simulator=simulator, objective="expectation")(gamma, beta)
+            for simulator in simulators_to_run_names_no_qiskit
+        ]
+        f2 = [
+            get_qaoa_sk_objective(N, p, J=J, parameterization="gamma beta", simulator=simulator, objective="expectation")(gamma, beta)
+            for simulator in simulators_to_run_names_no_qiskit
+        ]
+
+        print(f1, f2)
+
+        assert np.allclose(f1, f2)
+
+
+def test_sk_qaoa_obj_consistency_across_simulators():
+    N = 8
+    J = np.random.randn(N, N)
+    J = (J + J.T) / 2
+    np.fill_diagonal(J, 0)
+
+    for p in range(1, 18):
+        gamma, beta = get_sk_gamma_beta(p)
+        for objective in ["expectation", "overlap"]:
+            qaoa_objectives = [
+                get_qaoa_sk_objective(N, p, J=J, parameterization="gamma beta", simulator=simulator, objective=objective)(gamma, beta)
+                for simulator in simulators_to_run_names_no_qiskit
+            ]
+            assert np.all(np.isclose(qaoa_objectives, qaoa_objectives[0]))
+
+
+@pytest.mark.parametrize("simulator", simulators_to_run_names_no_qiskit)
+def test_sk_qaoa_obj_fixed_angles_and_precomputed_energies(simulator):
+    N = 10
+    J = np.random.randn(N, N)
+    J = (J + J.T) / 2
+    np.fill_diagonal(J, 0)
+    for max_p in [11, 4]:
```

Comment:
The purpose of this for loop is unclear to me; why vary max_p if running the inside of the loop with `max_p=4` corresponds to a subset of the run with `max_p=11`?

Reply:
Fixed!